packages/ptokens-assets-evm/src/ptokens-evm-provider.ts

```diff
@@ -205,24 +197,37 @@ export class pTokensEvmProvider implements pTokensAssetProvider {
         return false
       }
     }, _pollingTime)
+    if (!receipt) throw new Error('Polling function stopped unexpectedly. No receipt found')
```

Comment:
This does not apply if you use waitForTransactionReceipt. But, have you ever seen an undefined here? The polling doesn't exit until receipt has a value.

Reply:
This was to solve a TS error. In any case I remove the polling and use waitForTransactionReceipt.